server/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -557,6 +557,25 @@ public IndexMetaData getIndexSafe(Index index) {
         throw new IndexNotFoundException(index);
     }
 
+    /**
+     * Retrieves the write-index for a given <code>alias</code>
+     *
+     * @param alias the alias to search for a write-index with
+     * @return the write-index {@link Index} for <code>alias</code>, null if no write-index is configured
+     */
+    public Index getWriteIndex(String alias) {
```

Comment:
this is usage? I think we should remove it and use it in the follow up PRs we're planning?

Reply:
I will double check, but I believe it will be used by the index-name resolver when used for indexing. It is not used yet since we haven't implemented that yet